x-pack/plugins/maps/public/inspector/vector_tile_adapter/components/vector_tile_inspector.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { Adapters } from '@kbn/inspector-plugin/public';
+import { EuiComboBox, EuiComboBoxOptionOption, EuiSpacer, EuiTabs, EuiTab } from '@elastic/eui';
+import { EmptyPrompt } from './empty_prompt';
+import type { TileRequest } from '../types';
+import { TileRequestTab } from './tile_request_tab';
+import { RequestsViewCallout } from './requests_view_callout';
+
+interface Props {
+  adapters: Adapters;
+}
+
+interface State {
+  selectedLayer: EuiComboBoxOptionOption<string> | null;
+  selectedTileRequest: TileRequest | null;
+  tileRequests: TileRequest[];
+  layerOptions: Array<EuiComboBoxOptionOption<string>>;
+}
+
+class VectorTileInspector extends Component<Props, State> {
+  private _isMounted = false;
+
+  state: State = {
+    selectedLayer: null,
+    selectedTileRequest: null,
+    tileRequests: [],
+    layerOptions: [],
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._onAdapterChange();
+    this.props.adapters.vectorTiles.on('change', this._debouncedOnAdapterChange);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    this.props.adapters.vectorTiles.removeListener('change', this._debouncedOnAdapterChange);
+  }
+
+  _onAdapterChange = () => {
+    const layerOptions = this.props.adapters.vectorTiles.getLayerOptions() as Array<
+      EuiComboBoxOptionOption<string>
+    >;
+    if (layerOptions.length === 0) {
+      this.setState({
+        selectedLayer: null,
+        selectedTileRequest: null,
+        tileRequests: [],
+        layerOptions: [],
+      });
+      return;
+    }
+
+    const selectedLayer =
+      this.state.selectedLayer &&
+      layerOptions.some((layerOption) => {
+        return this.state.selectedLayer?.value === layerOption.value;
+      })
+        ? this.state.selectedLayer
+        : layerOptions[0];
+    const tileRequests = this.props.adapters.vectorTiles.getTileRequests(selectedLayer.value);
```

Comment:
When I zoom out as far as possible, the inspector shows some negative Y tiles. If we try to Open in Console, we get an expected error from Elasticsearch
![localhost_5601_fao_app_maps_map_5dd88580-1906-11e9-919b-ffe5949a18d2](https://user-images.githubusercontent.com/1638483/167211038-fc95a98e-f7cd-491c-9cb0-fc1f80d5278e.png)
.

Reply:
Thanks for pointing this out. I have adjusted the tile generation logic (and a unit test) to only return a single tile for zoom level 0.